docs/lib/auth/fragments/android/check_authentication_state/10_checkSignInState.md

```diff
@@ -0,0 +1,146 @@
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+Amplify.Auth.fetchAuthSession(
+    result -> {
+        AWSCognitoAuthSession cognitoAuthSession = (AWSCognitoAuthSession) result;
+            // Check refresh token is valid or not
+            switch (cognitoAuthSession.getUserPoolTokens().getType()) {
+                case SUCCESS:
+                    Log.i("AuthQuickStart", "Refresh token is valid.");
+                    break;
+                case FAILURE:
+                    Log.e("AuthQuickStart", "Refresh token expired. User must re-authenticate by signing in again.", cognitoAuthSession.getUserPoolTokens().getError());
+                    // Check whether AuthSession is expired
+                    subscribeToAuthEvents();
```

Comment:
why do we need to subscribe to auth events anyways? I was expecting at this point a comment like
```
// User is no longer authenticated, display the login screen for the user to re-authenticate
``` 

why don't we subscribe to auth events regardless of the result of `cognitoAuthSession.getUserPoolTokens().getType()` ?